lens-cube/src/main/java/org/apache/lens/cube/parse/StorageTableResolver.java

```diff
@@ -169,6 +152,25 @@ public void rewriteContext(CubeQueryContext cubeql) throws LensException {
     phase = phase.next();
```

Comment:
cubeql.setNonexistingParts(nonExistingPartitions) is required  for each phase ?  Might make sense only for 2/3